pkg/operator/baremetal_config.go

```diff
@@ -136,7 +170,7 @@ func getProvisioningDHCPRange(baremetalConfig BaremetalProvisioningConfig) *stri
 	// to be empty.
 	if baremetalConfig.ProvisioningDHCPRange != "" {
 		return &(baremetalConfig.ProvisioningDHCPRange)
-	} else if baremetalConfig.ProvisioningDHCPExternal {
+	} else if baremetalConfig.ProvisioningDHCPExternal || baremetalConfig.ProvisioningNetwork != provisioningNetworkManaged {
```

Comment:
Based on the logic above in getBaremetalProvisioningConfig, baremetalConfig.ProvisioningDHCPRange will never be empty, and this function looks like it's doing the wrong thing.  If there is a garbage value in baremetalConfig.ProvisioningDHCPRange (say '0.0.0.0') because that field is required and you're setting ProvisioningNetwork to unmanaged, you're going to get the garbage value and you probably want the nil here?  I can't really make sense of this function, but it seems like it's doing the wrong thing.

Reply:
Yes, it is confusing. The check is only looking to see if the DHCPRange is empty or not. I am not sure how we can check if it is a garbage value or not.

The function is doing this:
1. If DHCPrange is not empty, just return string as is.
2. If it is empty, then if the provisioningNetwork is not "Managed" , then return " ".
3. All other cases, return nil.

Reply:
I thought there was a reply to this according to my email, but it looks like github lost it?

Anyway, according to the getBaremetalProvisioningConfig function, baremetalConfig.ProvisioningDHCPRange is not allowed to be empty, so this function is never going to return anything other than the value of baremetalConfig.ProvisioningDHCPRange in actual use as far as I can tell.

Reply:
An empty string is a valid  value for the provisioningDHCPRange. When then provisioningDHCPExternal is true or when provisioningNetwork is in "Unmanaged' or "Disabled" state, provisioningDHCPRange is passed in as " ".

Reply:
Okay, I see what you're saying now.  I must have gotten confused.